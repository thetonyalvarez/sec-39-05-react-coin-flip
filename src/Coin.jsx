import React, { useState } from 'react'
import Box from '@mui/material/Box'
import './Coin.css'


const Coin = (props) => {
  let side;
  let coinImg;

  if (props.side == 0) {
    return (
      <Box></Box>
    )
  }

  if (props.side == 1) {
    side = 'heads'
  } else if (props.side == 2) {
    side = 'tails'
  }
  coinImg = `./src/${side}.png`

  return (
    <Box
      sx={{p: 4}}>
      <img
        className="CoinImg"
        src={coinImg}
        alt={side}
        data-testid="coin-img"
      />
    </Box>
  )
}

export default Coin