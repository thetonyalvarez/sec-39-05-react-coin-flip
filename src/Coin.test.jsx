import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { expect, it, describe } from 'vitest'
import Coin from "./Coin";

it("renders without crashing", function() {
  render(<Coin/>)
});

it("matches snapshot", function() {
  const {asFragment} = render(<Coin/>);
  expect(asFragment()).toMatchSnapshot();
});