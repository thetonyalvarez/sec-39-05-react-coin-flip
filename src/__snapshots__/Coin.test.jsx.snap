// Vitest Snapshot v1

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1wcaknn"
  >
    <img
      class="CoinImg"
      data-testid="coin-img"
      src="./src/undefined.png"
    />
  </div>
</DocumentFragment>
`;
